Search and highlight word in the DOM

Amin-Partovi opened the pull request:
This PR introduces a new hook called useHighlight. The useHighlight hook accepts two arguments: keyword (the word we want to search for) and isReady (a boolean argument indicating whether the search should be performed). It returns a ref that should be assigned to the container element we want to search within.
I am ready to receive your feedback and address any comments you may have on my PR. Your time and consideration are greatly appreciated.

imbhargav5 commented on packages/rooks/src/hooks/useHighlight.ts:
```diff
@@ -0,0 +1,35 @@
+ */
+import { useEffect, useRef } from "react";
+
+function useHighlight({
```
Super clever. I have a small preference for argument structure. Can we do something such as this. 

```tsx
useHighlight(keyword, {
   when: true
})
```

That way it will match some of the other hooks in our library. What do you think?

Amin-Partovi replied:
Yes, I modify it for the consistency

imbhargav5 commented on packages/rooks/src/hooks/useHighlight.ts:
```diff
@@ -0,0 +1,35 @@
+        const regex = new RegExp(keyword, "g");
+        const newText = text.replace(
+          regex,
+          `<mark class="highlighted">${keyword}</mark>`
```
I was wondering if we can somehow get the DOM from another option and default the DOM to `<mark class="highlighted">{}</mark>`

```tsx
useHighlight(keyword, {
    render:(match) => `<mark class="highlighted">${keyword}</mark>`
})
```

Amin-Partovi replied:
I wonder if I have understood your point correctly. I have added a render argument for cases where we want to include optional markup for the keywords.

imbhargav5 commented on packages/rooks/src/hooks/useHighlight.ts:
```diff
@@ -0,0 +1,35 @@
+        const regex = new RegExp(keyword, "g");
+        const newText = text.replace(
+          regex,
+          `<mark class="highlighted">${keyword}</mark>`
```
One thing I would like to add in the tests is that it should not highlight if it is already highlighted. For eg:
✅
`something keyword` => `something <mark class="highlighted">keyword</mark>` 
❌
`something <mark class="highlighted">keyword</mark>` => `something <mark class="highlighted"><mark class="highlighted">keyword</mark></mark>`

Amin-Partovi commented:
@imbhargav5 
Hi Bhargav,

I hope you're doing well. I have addressed the comments you provided, and I was wondering if you could find some time to review my PR. I'm open to fixing any issues you mentioned and making any necessary adjustments.

Thank you for your time and consideration
